io_stream.c

```diff
@@ -141,6 +169,53 @@ void* php_mongo_io_stream_connect(mongo_con_manager *manager, mongo_server_def *
 				return NULL;
 			}
 		} else {
+			zval **zcert;
+
+			if (php_stream_context_get_option(stream->context, "ssl", "peer_certificate", &zcert) == SUCCESS && Z_TYPE_PP(zcert) == IS_RESOURCE) {
+				zval **verify_peer_name, **verify_expiry;
+				int resource_type;
+				X509 *cert;
+				int type;
+
+
+				zend_list_find(Z_LVAL_PP(zcert), &resource_type);
+				cert = (X509 *)zend_fetch_resource(zcert TSRMLS_CC, -1, "OpenSSL X.509", &type, 1, resource_type);
+
+				if (!cert) {
+					mongo_manager_log(manager, MLOG_CON, MLOG_INFO, "Could not capture certificate of %s:%d", server->host, server->port);
+					php_stream_close(stream);
+					return NULL;
+				}
+
+#if PHP_VERSION_ID < 50600
+				/* This option is available since PHP 5.6.0 */
+				if (php_stream_context_get_option(options->ctx, "ssl", "verify_peer_name", &verify_peer_name) == SUCCESS && zend_is_true(*verify_peer_name)) {
+					if (php_mongo_verify_hostname(server, cert TSRMLS_CC) == FAILURE) {
+						*error_message = strdup("Hostname doesn't match");
+						php_stream_close(stream);
+						return NULL;
+					}
+					mongo_manager_log(manager, MLOG_CON, MLOG_INFO, "stream_connect: Valid peer name for %s:%d", server->host, server->port);
```

Comment:
Wouldn't we warn in the log if there is *no* valid peer certificate as well? Just like with the error_message that you return in 194?